### PR TITLE
docs: app.controller.foo instead of 'foo'

### LINKED ISSUE #920: 在使用typescript开发时，router的方法无法进行严格的typing定义
当在使用Typescript进行egg的开发的时候，无法对router的函数进行严格定义。
现在的函数定义是这样的
`app.verb([name:string],rule:string|regex, [middleware1, ..., middlewareN,] controllerAction:middleware|string);`

在koa-router中，action是通过middleware的方式定义，Egg对koa-router进行了扩展，提供了一个string的格式，但是这样做，就会在函数后面添加了一个必须的参数，而这个参数和前面的可选middleware数组类型不一致。  

 建议将app中route的定义改成把controller.action 放到middlewares之前，这样的话函数的定义会比较统一。typings可以这样定义
`app.verb([name:string], rule:string|regex, controllerAction:middleware|string, ...middlewares:middleware[])`

### PR DESCRIPTION
**Pull request by dead-horse**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change

closes #920

### CODE REVIEW

**Comment by dead-horse** [comment]
和业务方聊的时候他们也不太能理解直接传一个字符串，再加上中间有一堆的 middleware 也不能穿字符串，更加容易混乱。

File: docs/source/zh-cn/basics/router.md [hunk 1]
```diff
@@ -326,13 +327,13 @@ module.exports = app => {
...
-  app.get('/news/detail', 'news.detail');
+  app.get('/news/list', app.controller.news.list);
+  app.get('/news/detail', app.controller.news.detail);
 };
 
 // app/router/admin.js
 module.exports = app => {
-  app.get('/admin/user', 'admin.user');
-  app.get('/admin/log', 'admin.log');
+  app.get('/admin/user', app.controlleradmin.user);
```
Comment (atian25) [inline docs/source/zh-cn/basics/router.md]: 少了个点

**Comment by dead-horse** [comment]
quickstart 也改了

**Comment by atian25** [comment]
+1

**Pull request merged**